### ISSUE TITLE
Move our packages' CI to Ubuntu Focal

### ISSUE THREAD

**Issue by emersonknapp** [ros-tooling/aws-roadmap#247]
## Description

Foxy uses Focal - we need our CI for packages we're releasing to use it
* [x] rosbag2
* ~~rosbag2_bag_v2~~ tracked in https://github.com/ros-tooling/aws-roadmap/issues/265
* [x] libstatistics_collector
* [x] system_metrics_collector

As of this writing, Github Actions do not support native Focal runners. We will likely need to switch over to a docker-based CI for these packages to enable this

## Test Plan

* CI passes when using Focal

## Documentation Plan

N/A

## Release Plan

* Merge PR changing Action CI settings (when green)

## Acceptance Criteria

* [ ] Code has been implemented, reviewed and merged.
* [ ] Test plan has been completed
* [ ] Release plan has been completed

Once all above items are checked, this story can be moved to done.

## Resources

* [FIXME](http://...)

**Comment by thomas-moulard** [comment]
rosbag2 PR in review: https://github.com/ros2/rosbag2/pull/401

**Comment by thomas-moulard** [comment]
@prajakta-gokhale I think it's unreasonable to bundle rosbag2_bag_v2 in this user story, as this CI never worked. Would it be OK to have a follow-up task an extract the subpart of this user story which is more complex. Updating rosbag2_bag_v2 should be blocked on making the CI work first.

**Comment by prajakta-gokhale** [comment]
That's fine by me.

**Comment by prajakta-gokhale** [comment]
@thomas-moulard seems like the last PR from rosbag2 is merged. Is there anything else left for this task?
